eth/filters/filter_system.go

```diff
@@ -351,6 +377,20 @@ func (es *EventSystem) handleTxsEvent(filters filterIndex, ev core.NewTxsEvent)
 	}
 }
 
+func (es *EventSystem) handleFinalizedEvent(filters filterIndex, ev core.ChainEvent) {
+	if ev.FinalizedBlockNumber == 0 || es.lastFinalized == ev.FinalizedBlockNumber {
```

Comment:
this condition is not correct, if `ev.FinalizedBlockNumber > 0`, it still passes the condition no matter the second condition meet or not